### ISSUE TITLE
HCal 2017dev Geometry Built from XML Differs from its DB Dump

### ISSUE THREAD

**Issue by ianna** [cms-sw/cmssw#15985]
HCal 2017dev geometry in pre12 does not provide correct summary for DB storage.

 * The shapes are correct, but their placement (e.g. rotation in Phi) looks off in phi(Y?) by a factor of 10 and placement in Z seems to be abs(Z)

HCal reco geometry built from XML vs DB tag:
![screenshot 2016-09-22 12 07 19](https://cloud.githubusercontent.com/assets/1390682/18828044/56c0dce2-83d6-11e6-909c-eef311670a20.png)
![screenshot 2016-09-22 12 20 04](https://cloud.githubusercontent.com/assets/1390682/18828045/56c6bfe0-83d6-11e6-9be6-7060c403ec45.png)

HCal sim geometry built from XML vs DB tag:
![screenshot 2016-09-23 12 42 39](https://cloud.githubusercontent.com/assets/1390682/18828063/700cd160-83d6-11e6-8c02-641d5e2f7c13.png)
![screenshot 2016-09-23 12 46 50](https://cloud.githubusercontent.com/assets/1390682/18828064/70141e0c-83d6-11e6-80b3-95d51a93284f.png)

Comparing the shapes, for example, of a DetId HB(-16,1,1) it looks like corners order gets scrambled for the same shape definition based on dX, dY and dZ: 
```
*** Shape TGeoArb8: TGeoArb8 ***
    point #0 : x=   -7.78802 y=   -7.90493 z=   -4.68055
    point #1 : x=   -7.78802 y=    7.90493 z=   -4.68055
    point #2 : x=   24.63257 y=    7.90493 z=   -4.68055
    point #3 : x=   24.63257 y=   -7.90493 z=   -4.68055
    point #4 : x=  -25.47067 y=   -8.31365 z=    4.68055
    point #5 : x=  -25.47067 y=    8.31365 z=    4.68055
    point #6 : x=    8.62616 y=    8.31365 z=    4.68055
    point #7 : x=    8.62616 y=   -8.31365 z=    4.68055
 Bounding box:
*** Shape TGeoArb8: TGeoBBox ***
    dX =    25.05162
    dY =     8.31365
    dZ =     4.68055
    origin: x=   -0.41905 y=    0.00000 z=    0.00000
*** Shape TGeoArb8: TGeoArb8 ***
    point #0 : x=    7.78802 y=   -7.90493 z=   -4.68055
    point #1 : x=  -24.63257 y=   -7.90493 z=   -4.68055
    point #2 : x=  -24.63257 y=    7.90493 z=   -4.68055
    point #3 : x=    7.78802 y=    7.90493 z=   -4.68055
    point #4 : x=   25.47067 y=   -8.31365 z=    4.68055
    point #5 : x=   -8.62616 y=   -8.31365 z=    4.68055
    point #6 : x=   -8.62616 y=    8.31365 z=    4.68055
    point #7 : x=   25.47067 y=    8.31365 z=    4.68055
 Bounding box:
*** Shape TGeoArb8: TGeoBBox ***
    dX =    25.05162
    dY =     8.31365
    dZ =     4.68055
    origin: x=    0.41905 y=    0.00000 z=    0.00000
```

**Comment by ianna** [comment]
Similar problem with using old geometry loader:
![screenshot 2016-09-26 11 45 34](https://cloud.githubusercontent.com/assets/1390682/18829958/f0185d40-83de-11e6-98c4-09b2c7c15a6c.png)

**Comment by ianna** [comment]
@bsunanda - is it due to degree vs radian interpretation?

**Comment by mmusich** [comment]
@ianna, to understand this issue, is the problem involving only the dump or accessing the geometry from DB in CMSSW jobs yields different results than from XML?
@franzoni

**Comment by ianna** [comment]
@mmusich - no, nothing to do with DB access.

**Comment by mmusich** [comment]
@ianna, let me rephrase then. Accessing the geometry via:

 `--geometry Extended2017dev`

is completely equivalent than accessing via:

`--geometry DB:Extended` (if using the Global Tag `81X_upgrade2017_HCALdev_v2`, that has in it the following tag: `XMLFILE_Geometry_81YV8_Extended2017dev_mc`
?
Thanks for clarifications

**Comment by ianna** [comment]
@mmusich - accessing the geometry - yes, e.g. DDCompactView is identical. However, there is a difference in which code is invoked in building a sub-detector Reco geometry. The DDCompactView is not used in this case, especially if the reco geometry tags are present in GT. 

FYI, another issue with 2017 xml geometry is that there are important overlaps which cannot be ignored. Tracker pushed its envelope mother volume by changing constants which are used elsewhere. Unfortunately, this caused overlaps in a beam pipe within tracker region. Most probably, it will cause crashes. I'm working on eliminating these. No actual changes to Tracker reco geometry are foreseen - just changing its mother volumes made of Air.

**Comment by makortel** [comment]
May I ask what is the plan for the phase1 geometry for pre13? Is the problem expected to be solved by then, or do we just keep using the XML geometry until it is fixed?

**Comment by ianna** [comment]
@makortel - I've found a couple of bugs but the major problem is in broken contracts. I'm assessing if some class modifications can be rolled back.

**Comment by ianna** [comment]
* 2017dev from a local DB tag:
![screenshot 2016-10-14 12 21 55](https://cloud.githubusercontent.com/assets/1390682/19385609/b49dd47c-920f-11e6-9292-86774e3869a0.png)
* 2017 from a local DB tag:
![screenshot 2016-10-14 12 52 38](https://cloud.githubusercontent.com/assets/1390682/19385611/b4a0b5d4-920f-11e6-80cd-ef1db22fede5.png)
* Run1 from a DB GT tag:
![screenshot 2016-10-14 12 56 42](https://cloud.githubusercontent.com/assets/1390682/19385610/b4a084a6-920f-11e6-9ea6-9450163a1714.png)
